### PR TITLE
[tessellator]Expose and document smoothing approximation via the C/Dart API. Minor refactor of tessellator to supply fill type to method rather than to object.

### PR DESCRIPTION
**Pull request by dnfield**
See https://github.com/flutter/flutter/issues/99769

This allows clients of the tessellation API to specify different smoothing values to obtain more vertices for curves that will be drawn at scale.

Adds some docs, updates a link, and moves the Tessellator fill type to the method (partly in preparation for a second method on Tessellator to tessellate strokes).

Updates the IMPELLER_API macro to work on Windows (although it's currently very difficult to build on windows since a bunch of the build rules lead to errors when not on macOS).

@jonahwilliams FYI

### CODE REVIEW

File: geometry/path_component.h [hunk 1]
```diff
@@ -12,10 +12,31 @@
...
+  /// The scaling coefficient to use when translating to screen coordinates.
+  ///
+  /// Values approaching 0.0 will generate smoother looking curves with a
+  /// greater number of vertices, and will be more expensive to calculate.
   Scalar scale;
+
+  /// The tolerance value in radians for calculating sharp angles.
+  ///
+  /// Values approaching 0.0 will provide more accurate approximation of sharp
+  /// turns. A 0.0 vlaue means angle conditions are not considered at all.
```
Comment (jonahwilliams) [inline geometry/path_component.h:30]:
```suggestion
  /// turns. A 0.0 value means angle conditions are not considered at all.
```

File: tessellator/dart/lib/tessellator.dart [hunk 2]
```diff
@@ -7,6 +7,53 @@ import 'dart:ffi' as ffi;
...
+  /// The scaling coefficient to use when translating to screen coordinates.
+  ///
+  /// Values approaching 0.0 will generate smoother looking curves with a
+  /// greater number of vertices, and will be more expensive to calculate.
+  final double scale;
+
+  /// The tolerance value in radians for calculating sharp angles.
+  ///
+  /// Values approaching 0.0 will provide more accurate approximation of sharp
+  /// turns. A 0.0 vlaue means angle conditions are not considered at all.
```
Comment (jonahwilliams) [inline tessellator/dart/lib/tessellator.dart:47]:
```suggestion
  /// turns. A 0.0 value means angle conditions are not considered at all.
```

**Comment by jonahwilliams** [comment]
I don't see the IMPELLER_API changes, did you mean to upload them in this PR?

File: geometry/path_component.h (hunk 1 above)
Comment (dnfield) [reply geometry/path_component.h]: Ah, thanks, fixed these.

**Comment by dnfield** [comment]
My push attempt got messed up and Ididn't notice. It should be there now.

**Review by chinmaygarde — APPROVED** [review]

**Pull request merged**